### DISCUSSION TITLE
Unable to re-render conditional rules without using private JS method

### DISCUSSION THREAD

**Discussion by kennethormandy** [solspace/craft-freeform#1113] [Questions & Answers] [answered]
**Describe the bug or issue you're experiencing**

Hi there! As far as I’m able to tell, I’m not able to apply conditional rules again, if I modify a form externally. If an input with conditional rules is modified, the rules rules don’t seem to be applied again, until that field is interacted with by an end user.

I can’t seem to find a different approved way to do this.

**Steps to reproduce**

Here’s roughly what I’m hoping to accomplish:

1. Create a field with conditional rules, ex. only show another field when `myRadioField` is a certain value
2. Change the value of `myRadioField` when you click a button separate from Freeform, only after the `freeform-ready`, ex:
  ```js
  let myButton = document.getElementById("my-button");

  form.addEventListener("freeform-ready", (e) => {
    let form = e.freeform;
    let freeform = e.freeform.freeform;

    // Change an input with conditional rules, based on an external factor
    myButton.addEventListener("click", (evt) => {
      form.elements["myRadioField"].value = evt.target.value;

      // Conditional rules aren’t applied until that field is
      // interacted with, or this private method is used
      freeform._resetHandlers();
    });
  });
  ```

**Expected behavior**

This either happens automatically, or is possible to handle manually by using something public similar to `_resetHandlers()`.

**Craft & Plugin Info (please complete the following information):**

As far as I can tell, this is the same in Freeform 5, but I can certainly upgrade if I’m misunderstanding something there.

- Craft Version: 4.8.x
- Freeform Version: 4.1.15.1
- Freeform Edition: Pro
- Fresh Install or Upgrade:

Thanks very much!

**Comment by gustavs-gutmanis** [comment, marked as answer]
Hi @kennethormandy,

You're modifying the value of the field directly, but the rule handlers are listening for `change` and `click` events.
You could try triggering a `click` event on the radio field after you have modified its value, by doing this:

```js
  myButton.addEventListener("click", (evt) => {
    const field = form.elements["myRadioField"];
    field.value = evt.target.value;

    // Trigger the click event
    field.click();
  });
```

As a note, you might need to set the `checked` property to true on the radio button that is the one that has to be selected. So keep that in mind.

**Reply by kennethormandy** [reply]
Thanks, this got me on the right path. It works for radios, but it doesn’t seem to work for text inputs until I used a key up event.

Instead of `field.click()`, this seems to be working, but feels excessive:

```js
['click', 'change', 'keyup'].forEach(listener => {
  let event = new Event(listener)
  field.dispatchEvent(event)  
})
```

…because I think I’m really just trying to trigger the rule handler reload here?

v5: https://github.com/solspace/craft-freeform/blob/38532f02a9bf3d4677d45245e582b1e6f0b38a76/packages/scripts/src/lib/plugin/handlers/form/rules.ts#L77-L105

v4: https://github.com/solspace/craft-freeform/blob/d2cc149cb6cff60ca00ba3d665d274b69a2f8c60/packages/scripts/src/lib/plugin/handlers/form/rule-set.js#L72-L96
